sklearn/cluster/k_means_.py

```diff
@@ -165,9 +165,24 @@ def _tolerance(X, tol):
     return np.mean(variances) * tol
 
 
-def k_means(X, n_clusters, init='k-means++', precompute_distances='auto',
-            n_init=10, max_iter=300, verbose=False,
-            tol=1e-4, random_state=None, copy_x=True, n_jobs=1,
+def _check_sample_weights(X, sample_weights):
+    """Set sample_weights if None, and check for correct dtype"""
+    n_samples = X.shape[0]
+    if sample_weights is None:
+        return np.ones(n_samples, dtype=X.dtype)
+    else:
+        # verify that the number of samples is equal to the number of weights
+        if n_samples != len(sample_weights):
```

Comment:
We have a helper called check_consistent_length

Reply:
Tom made the same remark earlier, and I replied that the error message will then be very generic. I'd prefer the more specific error message, but happy to change.